examples/models/ResNet50.cpp

```diff
@@ -0,0 +1,1293 @@
+/***************************************************************************
+ *  Copyright 2020 Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ **************************************************************************/
+
+
+#include <assert.h>
+#include <chrono>
+#include <fstream>
+#include <iomanip>
+#include <memory>
+#include <vector>
+#include "dnnl.hpp"
+#include "example_utils.hpp"
+
+#include <cassert>
+#include <random>
+#include <string>
+#include <unordered_map>
+
+#include <stdio.h>
+
+using namespace dnnl;
+using tag = memory::format_tag;
+using dt = memory::data_type;
+
+memory::dim BATCH = 1;
```

Comment:
Why have this as a global variable? It is strange to see at the very end that this is a variable updated in a for loop. Could this not be a normal parameter passed to functions?

Reply:
I agree with your suggestion. The code was added as a quick fix to support multiple batch sizes. 
However, this is fixed correctly now.